digest.go

```diff
@@ -109,18 +109,11 @@ func (d Digest) Validate() error {
 		return ErrDigestInvalidFormat
```

Comment:
> And who knows, `!!!` could be a valid encoding for `foobar`.

Dropping the digest regexp is going to make this package much more permissive than image spec (which [has a number of charset restrictions][1]).  That's fine, go-digest is free to be more permissive, but it would be nice if there was a way to ask “is this a valid image-spec digest?”. “Does go-digest have any personal beef with this?” is a useful question, but potentially less useful for folks using go-digest to handle image-spec digests.

[1]: https://github.com/opencontainers/image-spec/blob/3437101f0c0ed264a53193cdec3b2ef6f7838f2c/descriptor.md#digests